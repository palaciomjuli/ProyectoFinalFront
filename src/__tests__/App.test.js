const { render, cleanup } = require('@testing-library/react');
const { default: App } = require('../App');
const { ContextProvider } = require('../Components/utils/global.context');
const { MemoryRouter } = require('react-router-dom');

const mockedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedNavigate,
}));

const context = {
  isFetching: false,
  isDarkMode: true,
  data: [],
  favorites: [],
};

describe('Test on <App />', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    cleanup();
  });

  test('should match snaptshot', () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/home']}>
        <ContextProvider value={context}>
          <App />
        </ContextProvider>
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });
});
