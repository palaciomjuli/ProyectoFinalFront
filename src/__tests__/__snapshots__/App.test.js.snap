// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <App /> should match snaptshot 1`] = `
<div>
  <div
    class="app"
  >
    <header
      class="dark header"
    >
      <h1
        aria-label="h1"
        class="header__h1"
      >
        DH Odonto
      </h1>
      <nav
        class="navbar"
      >
        <a
          aria-current="page"
          class="active"
          href="/home"
        >
          Home
        </a>
        <a
          class=""
          href="/contact"
        >
          Contact
        </a>
        <a
          class=""
          href="/favs"
        >
          Favs
        </a>
        <button
          aria-label="themeButton"
          class="menu-modal__theme-button"
        >
          <svg
            class="bi bi-brightness-high-fill"
            fill="white"
            viewBox="0 0 16 16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 8a4 4 0 11-8 0 4 4 0 018 0zM8 0a.5.5 0 01.5.5v2a.5.5 0 01-1 0v-2A.5.5 0 018 0zm0 13a.5.5 0 01.5.5v2a.5.5 0 01-1 0v-2A.5.5 0 018 13zm8-5a.5.5 0 01-.5.5h-2a.5.5 0 010-1h2a.5.5 0 01.5.5zM3 8a.5.5 0 01-.5.5h-2a.5.5 0 010-1h2A.5.5 0 013 8zm10.657-5.657a.5.5 0 010 .707l-1.414 1.415a.5.5 0 11-.707-.708l1.414-1.414a.5.5 0 01.707 0zm-9.193 9.193a.5.5 0 010 .707L3.05 13.657a.5.5 0 01-.707-.707l1.414-1.414a.5.5 0 01.707 0zm9.193 2.121a.5.5 0 01-.707 0l-1.414-1.414a.5.5 0 01.707-.707l1.414 1.414a.5.5 0 010 .707zM4.464 4.465a.5.5 0 01-.707 0L2.343 3.05a.5.5 0 11.707-.707l1.414 1.414a.5.5 0 010 .708z"
            />
          </svg>
        </button>
      </nav>
      <button
        class="header__button"
      >
        <svg
          class="bi bi-list"
          fill="#fff"
          height="30"
          viewBox="0 0 16 16"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2.5 12a.5.5 0 01.5-.5h10a.5.5 0 010 1H3a.5.5 0 01-.5-.5zm0-4a.5.5 0 01.5-.5h10a.5.5 0 010 1H3a.5.5 0 01-.5-.5zm0-4a.5.5 0 01.5-.5h10a.5.5 0 010 1H3a.5.5 0 01-.5-.5z"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </header>
    <main
      class="main mainDark"
    />
    <footer>
      <a
        class="footer dark"
        href="https://www.digitalhouse.com/"
        rel="noreferrer"
        target="_blank"
      >
        <p
          class="footer__p"
        >
          Powered by
        </p>
        <img
          alt="DH-logo"
          src="DH.ico"
          style="width: 40px; margin-left: 10px; height: 40px;"
        />
      </a>
    </footer>
  </div>
</div>
`;
